app/src/main/java/de/westnordost/streetcomplete/overlays/surface/PathSurfaceOverlayForm.kt

```diff
@@ -0,0 +1,385 @@
+            val selectedMainSurfaceNoteText = inState.getString(SELECTED_MAIN_SURFACE_NOTE_TEXT)
+            val selectedCyclewaySurfaceIndex = inState.getInt(SELECTED_CYCLEWAY_SURFACE_INDEX)
+            val selectedCyclewaySurfaceNoteText = inState.getString(SELECTED_CYCLEWAY_SURFACE_NOTE_TEXT)
+            val selectedFootwaySurfaceIndex = inState.getInt(SELECTED_FOOTWAY_SURFACE_INDEX)
+            val selectedFootwaySurfaceNoteText = inState.getString(SELECTED_FOOTWAY_SURFACE_NOTE_TEXT)
+            selectedStatusForMainSurface = if (selectedMainSurfaceIndex != -1) items[selectedMainSurfaceIndex] else null
+            if (selectedMainSurfaceNoteText != null) {
+                binding.explanationInputMainSurface.text = SpannableStringBuilder(selectedMainSurfaceNoteText)
+            }
+            selectedStatusForCyclewaySurface = if (selectedCyclewaySurfaceIndex != -1) items[selectedCyclewaySurfaceIndex] else null
+            if (selectedCyclewaySurfaceNoteText != null) {
+                binding.explanationInputCyclewaySurface.text = SpannableStringBuilder(selectedCyclewaySurfaceNoteText)
+            }
+            selectedStatusForFootwaySurface = if (selectedFootwaySurfaceIndex != -1) items[selectedFootwaySurfaceIndex] else null
+            if (selectedFootwaySurfaceNoteText != null) {
+                binding.explanationInputFootwaySurface.text = SpannableStringBuilder(selectedFootwaySurfaceNoteText)
+            }
+        }
+
+        override fun onSaveInstanceState(outState: Bundle) {
+            super.onSaveInstanceState(outState)
+            outState.putInt(SELECTED_MAIN_SURFACE_INDEX, items.indexOf(selectedStatusForMainSurface))
+            outState.putString(SELECTED_MAIN_SURFACE_NOTE_TEXT, noteText())
+            outState.putInt(SELECTED_CYCLEWAY_SURFACE_INDEX, items.indexOf(selectedStatusForCyclewaySurface))
+            outState.putString(SELECTED_CYCLEWAY_SURFACE_NOTE_TEXT, noteText())
+            outState.putInt(SELECTED_FOOTWAY_SURFACE_INDEX, items.indexOf(selectedStatusForFootwaySurface))
+            outState.putString(SELECTED_FOOTWAY_SURFACE_NOTE_TEXT, noteText())
+        }
+
+        /* -------------------------------------- apply answer -------------------------------------- */
+
+        override fun isFormComplete(): Boolean {
+            if (selectedStatusForMainSurface == null) {
+                if (selectedStatusForCyclewaySurface == null || selectedStatusForFootwaySurface == null) {
+                    return false
+                }
+            }
+            return hasChanges()
+        }
+
+    fun noteText(): String? {
+        return binding.explanationInputMainSurface.nonBlankTextOrNull
+    }
+
+    private fun cyclewayNoteText(): String? {
+        return binding.explanationInputCyclewaySurface.nonBlankTextOrNull
+    }
+
+    private fun footwayNoteText(): String? {
+        return binding.explanationInputFootwaySurface.nonBlankTextOrNull
+    }
```

Comment:
I think these should be `val`s with a getter instead:

```suggestion
    val noteText get() = binding.explanationInputMainSurface.nonBlankTextOrNull
    private val cyclewayNoteText get() = binding.explanationInputCyclewaySurface.nonBlankTextOrNull
    private val footwayNoteText get() = binding.explanationInputFootwaySurface.nonBlankTextOrNull
```